paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSheetViewModel.kt

```diff
@@ -516,64 +589,49 @@ internal class CustomerSheetViewModel(
-                savedPaymentMethods = currentViewState.savedPaymentMethods,
+                isLiveMode = isLiveModeProvider(),
+                cbcEligibility = customerState.cbcEligibility,
+                savedPaymentMethods = emptyList(),
                 allowsRemovalOfLastSavedPaymentMethod = configuration.allowsRemovalOfLastSavedPaymentMethod,
-                // TODO(samer-stripe): Set this based on customer_session permissions
-                canRemovePaymentMethods = true,
+                canRemovePaymentMethods = customerState.canRemove,
             )
         )
     }
 
     private fun removePaymentMethodFromState(paymentMethod: PaymentMethod) {
-        val currentViewState = viewState.value
-        val newSavedPaymentMethods = currentViewState.savedPaymentMethods.filter { it.id != paymentMethod.id!! }
-
-        if (currentViewState is CustomerSheetViewState.SelectPaymentMethod) {
-            updateViewState<CustomerSheetViewState.SelectPaymentMethod> { viewState ->
-                val originalSelection = originalPaymentSelection
+        val currentCustomerState = customerState.value
+        val newSavedPaymentMethods = currentCustomerState.paymentMethods.filter { it.id != paymentMethod.id!! }
 
-                val didRemoveCurrentSelection = viewState.paymentSelection is PaymentSelection.Saved &&
-                    viewState.paymentSelection.paymentMethod.id == paymentMethod.id
+        val currentSelection = currentCustomerState.currentSelection
+        val originalSelection = originalPaymentSelection
 
-                val didRemoveOriginalSelection = viewState.paymentSelection is PaymentSelection.Saved &&
-                    originalSelection is PaymentSelection.Saved &&
-                    viewState.paymentSelection.paymentMethod.id == originalSelection.paymentMethod.id
-
-                if (didRemoveOriginalSelection) {
-                    originalPaymentSelection = null
-                }
+        val didRemoveCurrentSelection = currentSelection is PaymentSelection.Saved &&
+            currentSelection.paymentMethod.id == paymentMethod.id
 
-                val updatedStateCanUpdate = canEdit(
-                    viewState.allowsRemovalOfLastSavedPaymentMethod,
-                    newSavedPaymentMethods,
-                    viewState.cbcEligibility
-                )
+        val didRemoveOriginalSelection = currentSelection is PaymentSelection.Saved &&
+            originalSelection is PaymentSelection.Saved &&
+            currentSelection.paymentMethod.id == originalSelection.paymentMethod.id
 
-                viewState.copy(
-                    savedPaymentMethods = newSavedPaymentMethods,
-                    paymentSelection = viewState.paymentSelection.takeUnless {
-                        didRemoveCurrentSelection
-                    } ?: originalPaymentSelection,
-                    isEditing = viewState.isEditing && updatedStateCanUpdate
```

Comment:
`isEditing` is now managed above in `StateFlow` and `init` where it is collected.